nuplan_garage/planning/simulation/planner/pdm_planner/abstract_pdm_closed_planner.py

```diff
@@ -0,0 +1,178 @@
+        self._generator = PDMGenerator(trajectory_sampling, proposal_sampling)
+        self._simulator = PDMSimulator(proposal_sampling)
+        self._scorer = PDMScorer(proposal_sampling)
+        self._emergency_brake = PDMEmergencyBrake(trajectory_sampling)
+
+        # lazy loaded
+        self._proposal_manager: Optional[PDMProposalManager] = None
+
+    def _update_proposal_manager(self, ego_state: EgoState):
+        """
+        Updates or initializes PDMProposalManager class
+        :param ego_state: state of ego-vehicle
+        """
+
+        current_lane = self._get_starting_lane(ego_state)
+
+        # TODO: Find additional conditions to trigger re-planning
+        create_new_proposals = self._iteration == 0
+
+        if create_new_proposals:
+            proposal_paths: List[PDMPath] = self._get_proposal_paths(current_lane)
+
+            self._proposal_manager = PDMProposalManager(
+                lateral_proposals=proposal_paths,
+                longitudinal_policies=self._idm_policies,
+            )
+
+        # update proposals
+        self._proposal_manager.update(current_lane.speed_limit_mps)
+
+    def _get_proposal_paths(self, current_lane: LaneGraphEdgeMapObject) -> List[PDMPath]:
+        """
+        Returns a list of path's to follow for the proposals. Inits a centerline.
+        :param current_lane: current or starting lane of path-planning
+        :return: lists of paths (0-index is centerline)
+        """
+        centerline_discrete_path = self._get_discrete_centerline(current_lane)
+        self._centerline = PDMPath(centerline_discrete_path)
+
+        # 1. save centerline path (necessary for progress metric)
+        output_paths: List[PDMPath] = [self._centerline]
+
+        # 2. add additional paths with lateral offset of centerline
+        if self._lateral_offsets is not None:
+            for lateral_offset in self._lateral_offsets:
+                offset_discrete_path = parallel_discrete_path(
+                    discrete_path=centerline_discrete_path, offset=lateral_offset
+                )
+                output_paths.append(PDMPath(offset_discrete_path))
+
+        return output_paths
+
+    def _get_closed_loop_trajectory(
+        self,
+        current_input: PlannerInput,
+    ) -> InterpolatedTrajectory:
+        """
+        Creates the closed-loop trajectory for PDM-Closed planner. 
+        :param current_input: planner input
+        :return: trajectory
+        """        
+
+        ego_state, observation = current_input.history.current_state
+
+        # 1. Environment forecast and observation update
+        self._observation.update(
+            ego_state, observation, current_input.traffic_light_data, self._route_lane_dict
+        )
+
+        # 2. Centerline extraction and proposal update
+        self._update_proposal_manager(ego_state)
+
+        # 3. Generate/Unroll proposals
+        proposals_array = self._generator.generate_proposals(
```

Comment:
In the current code its impossible to reuse the PDM-Closed planner to follow different proposals.
I think this would increase reusability and could be achieved by some minor refactoring.
However, I propose to do this in a follow-up PR